R/ops-addition.r

```diff
@@ -58,7 +57,11 @@ add_period_to_date <- function(per, date) {
   if (is.Date(date) && sum(new$sec, new$min, new$hour, na.rm = TRUE) != 0)
```

Comment:
As above, I think it would be better to decide on whether or not to return a date based on the values of the period. Do you want to have a go at making this change?

Reply:
I do.

Reply:
Just to be clear: so `as_date(NA) + seconds(1))` should also be a datetime, not a date. This is broader than infinites. New Issue?

Reply:
Right